Formula/cri-tools.rb

```diff
@@ -0,0 +1,34 @@
+class CriTools < Formula
+  desc "CLI and validation tools for Kubelet Container Runtime Interface (CRI)"
+  homepage "https://github.com/kubernetes-sigs/cri-tools"
+  url "https://github.com/kubernetes-sigs/cri-tools/archive/v1.22.0.tar.gz"
+  sha256 "76fc230a73dd7e8183f499c88effaf734540808f2f90287031a85d0a4d8512d9"
+  license "Apache-2.0"
+  head "https://github.com/kubernetes-sigs/cri-tools.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["BINDIR"] = bin
+
+    if build.head?
+      system "make", "install"
+    else
+      system "make", "install", "VERSION=#{version}"
+    end
+
+    output = Utils.safe_popen_read("#{bin}/crictl", "completion", "bash")
+    (bash_completion/"crictl").write output
+
+    output = Utils.safe_popen_read("#{bin}/crictl", "completion", "zsh")
+    (zsh_completion/"_crictl").write output
+
+    output = Utils.safe_popen_read("#{bin}/crictl", "completion", "fish")
+    (fish_completion/"crictl.fish").write output
+  end
+
+  test do
+    assert_match "crictl version", shell_output("#{bin}/crictl --version 2>&1")
+    assert_match "critest version", shell_output("#{bin}/critest --version 2>&1")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
It looks like `critest` is easy to test thanks to its dry-run mode.

On the other hand, just like any other `...ctl` tool, `crictl` is a bit more tricky. Tentatively testing if it successfully fails in https://github.com/Homebrew/homebrew-core/pull/88744/commits/c704754dbe49054d77862db020655c204fc79e16.

> If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is preferred over mocking a dependency.